luigi/scheduler.py

```diff
@@ -591,6 +612,7 @@ def get_work(self, worker, host=None, **kwargs):
                               for worker in self._state.get_active_workers())
 
         tasks = list(self._state.get_pending_tasks())
+        self._state.cascade_disabled_state(tasks)
```

Comment:
Hmm I haven't seen tasks go from disabled back to pending again, but then maybe I wasn't looking for it. Is there a config setting that controls the number of times this can happen, and the delay for making them pending again? I assume after it's flapped a couple of times it's just set to failed?

> Tasks also go from disabled to pending if you schedule a task as disabled and later schedule it as pending

How can you schedule a task as disabled and later schedule it as pending?

> It seems that the logic you'd actually want for this is to not count upstream disabled jobs as pending for the purposes of keep alive.

Yes, that's what my patch does. But it then propagates up through the graph so that the root nodes will be set to disabled if any of their children are disabled and everything else isn't pending.

Do you have an alternative suggestion for how to implement this? Perhaps adding a config setting to enable this behaviour (and default it to disabled) would be good so it doesn't affect people with current deployments.